pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -19,12 +19,175 @@ package kuberuntime
 import (
 	"fmt"
 	"io"
+	"os"
+	"path"
 
+	"github.com/golang/glog"
 	"k8s.io/kubernetes/pkg/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/types"
 	"k8s.io/kubernetes/pkg/util/term"
 )
 
+// getContainerLogsPath gets log path for container.
+func getContainerLogsPath(containerName string, podUID types.UID) string {
+	return path.Join(podLogsRootDirectory, string(podUID), fmt.Sprintf("%s.log", containerName))
+}
+
+// generateContainerConfig generates container config for kubelet runtime api.
+func (m *kubeGenericRuntimeManager) generateContainerConfig(container *api.Container, pod *api.Pod, restartCount int, podIP string) (*runtimeApi.ContainerConfig, error) {
+	opts, err := m.runtimeHelper.GenerateRunContainerOptions(pod, container, podIP)
+	if err != nil {
+		return nil, err
+	}
+
+	_, containerName, cid := buildContainerName(pod, container)
```

Comment:
First, it's not an implicit contract because users (i.e., kubelet here) has the freedom to choose a name it wants for the container, and runtime shouldn't care about it.

Second, we are trying to leverage labels more. However, we still want the name to convey *some* information so that it's user friendly. We can probably shorten the name but still have some information embedded in it.  I think the main reason that we are keeping the name as it is is to *not* intentionally deviate from the current implementation too much. If we ever need to ensure backward compatibility (i.e., in-place update of kubelet without draining the pods) down the road, we would still be able to make it happen somehow by padding more code. This makes the naming overhaul not desirable right now.